core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraFromRjsBuilder.kt

```diff
@@ -178,23 +214,30 @@ class RawInfraFromRjsBuilderImpl : RawInfraBuilder {
         return physicalSignalPool.add(builder.build())
     }
 
-    override fun trackSection(name: String?, init: TrackSectionBuilder.() -> Unit): TrackSectionId {
-        val builder = TrackSectionBuilderImpl(name)
-        builder.init()
-        return trackSectionPool.add(builder.build())
+    fun trackSection(name: String, chunks: StaticIdxList<TrackChunk>): TrackSectionId {
+        val sectionIdx = trackSectionPool.add(TrackSectionDescriptor(name, chunks))
+
+        sectionNameToIdxMap[name] = sectionIdx
+
+        val sectionOffsetChunks = TreeMap<Distance, TrackChunkId>()
```

Comment:
We have `DistanceRangeMap<T>`, it's supposed to be similar to a `TreeMap<Distance, T>` with a much smaller memory footprint. It doesn't necessarily matter if the object isn't meant to be kept after the loading process though, and it may be less flexible.

Reply:
The object isn't kept after loading, and I don't have a better tradeoff for the use here, but let's briefly discuss it today.